detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/parse/BitbakeRecipesParser.java

```diff
@@ -35,17 +41,20 @@
         }
 
         if (currentRecipe != null) {
-            bitbakeRecipes.add(currentRecipe);
+            bitbakeRecipes.put(currentRecipe.getName(), currentRecipe.getLayerNames());
+            if (currentRecipe.getLayerNames() != null) {
+                layerNames.addAll(currentRecipe.getLayerNames());
+            }
         }
 
-        return bitbakeRecipes;
+        return new ShowRecipesResults(layerNames, bitbakeRecipes);
     }
 
-    private BitbakeRecipe parseLine(String line, BitbakeRecipe currentRecipe, List<BitbakeRecipe> bitbakeRecipes) {
+    private BitbakeRecipe parseLine(String line, BitbakeRecipe currentRecipe, Map<String, List<String>> bitbakeRecipes) {
         if (line.contains(":") && !line.startsWith("  ")) {
             // Parse beginning of new component
             if (currentRecipe != null) {
-                bitbakeRecipes.add(currentRecipe);
+                bitbakeRecipes.put(currentRecipe.getName(), currentRecipe.getLayerNames());
```

Comment:
Why do we not want to add the `currentRecipe` to the `bitbakeRecipes` map in the other else-if statements in this method?

Reply:
I don't know... all I did in that method was change that list to a map. It's been that way for a while and I was unaware of a reason to question it.